looking-glass-client build failure

## Issue description
```$ sudo nixos-rebuild switch                                                                                                                                                                                                                                                                                                  
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/ygr0f5giydw8virjy14vscllh4ncff6s-looking-glass-client-a12.drv
  /nix/store/ljm1wapslzwfs9zrai78fax9cf4s5v7s-system-path.drv
  /nix/store/5bkddm6y52kxywsb5191lwjpk9qqxgdk-unit-systemd-fsck-.service.drv
  /nix/store/bs3g1dgsd8kz7faa880p6h1jirgsb85y-unit-accounts-daemon.service.drv
  /nix/store/f4cxds166hfhiyhh94wgpfkbdcn0kxzh-dbus-1.drv
  /nix/store/n4zrki7b5m1i5np873hcpxf1s4agili0-unit-dbus.service.drv
  /nix/store/wighhqqw9xhg75jpzwcjmv608kal3cw2-unit-polkit.service.drv
  /nix/store/90i24pkqfcbnmwmk2z6hypk7r49pkjx1-system-units.drv
  /nix/store/pq0a3v4bflki5ysf1cczakbkajdm81cj-unit-dbus.service.drv
  /nix/store/f6y8j95fdl79d7nfyih8igi2cxxhdm71-user-units.drv
  /nix/store/9vdnary7r52z33xbjcqab6pvlvc07dr6-xsession.drv
  /nix/store/l3pc6cczqy26sqc3qafd2r6ppagqj777-xsession-wrapper.drv
  /nix/store/pikfjpcypbm0mlr9qwr9sf7dyg448day-desktops.drv
  /nix/store/zpm14gb4jpn2cpjryb51zrfxq1v5bvav-lightdm.conf.drv
  /nix/store/6xv7054w05hv9ffyzzw32kpwxznbnw9h-etc.drv
  /nix/store/xjqrnrs0r8n4jwb0d2bsyja0mxr4zi44-nixos-system-nixos-workstation-19.09pre177651.aeb464dfd37.drv
building '/nix/store/ygr0f5giydw8virjy14vscllh4ncff6s-looking-glass-client-a12.drv'...
unpacking sources
unpacking source archive /nix/store/if4yklaww60kvb9xh555njln3ppzpszd-source
source root is source/client
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/aiv35zdk7ssd9psdpvvn5145cg2bv798-looking-glass-client-a12/include -DCMAKE_INSTALL_LIBDIR=/nix/store/aiv35zdk7ssd9psdpvvn5145cg2bv798-looking-glass-client-a12/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/aiv35zdk7ssd9psdpvvn5145cg2bv798-looking-glass-client-a12/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_DEPLOYMENT_TARGET= -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/1kl6ms8x56iyhylb2r83lq7j3jbnix7w-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/1kl6ms8x56iyhylb2r83lq7j3jbnix7w-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/1kl6ms8x56iyhylb2r83lq7j3jbnix7w-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/aiv35zdk7ssd9psdpvvn5145cg2bv798-looking-glass-client-a12
-- The C compiler identification is GNU 7.4.0
-- Check for working C compiler: /nix/store/w2px3blinprrzx6wva8x7m0l9l27jzj9-gcc-wrapper-7.4.0/bin/gcc
-- Check for working C compiler: /nix/store/w2px3blinprrzx6wva8x7m0l9l27jzj9-gcc-wrapper-7.4.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /nix/store/7s6n4qpbsiav8s8as35xmh27ssbh1nrz-pkg-config-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for modules 'sdl2;SDL2_ttf;gl;glu;egl;spice-protocol;fontconfig;x11;libconfig;nettle;hogweed'
--   Found sdl2, version 2.0.9
--   Found SDL2_ttf, version 2.0.15
--   Found gl, version 18.3.4
--   Found glu, version 9.0.0
--   Found egl, version 18.3.4
--   Found spice-protocol, version 0.12.15
--   Found fontconfig, version 2.12.6
--   Found x11, version 1.6.7
--   Found libconfig, version 1.7.2
--   Found nettle, version 3.4.1
--   Found hogweed, version 3.4.1
Package expat was not found in the pkg-config search path.
Perhaps you should add the directory containing `expat.pc'
to the PKG_CONFIG_PATH environment variable
Package 'expat', required by 'fontconfig', not found
Package expat was not found in the pkg-config search path.
Perhaps you should add the directory containing `expat.pc'
to the PKG_CONFIG_PATH environment variable
Package 'expat', required by 'fontconfig', not found
Package expat was not found in the pkg-config search path.
Perhaps you should add the directory containing `expat.pc'
to the PKG_CONFIG_PATH environment variable
Package 'expat', required by 'fontconfig', not found
Package expat was not found in the pkg-config search path.
Perhaps you should add the directory containing `expat.pc'
to the PKG_CONFIG_PATH environment variable
Package 'expat', required by 'fontconfig', not found
-- GMP libs: /nix/store/9b6qi566l7icp4z5l0fn86zfcbw8ws3f-gmp-6.1.2/lib/libgmp.so /nix/store/9b6qi566l7icp4z5l0fn86zfcbw8ws3f-gmp-6.1.2/lib/libgmpxx.so
-- Found GMP: /nix/store/bq3wh9rpy4aqahz24i4z0brmy46lg6yj-gmp-6.1.2-dev/include
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_CXX_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR


-- Build files have been written to: /build/source/client/build
building
build flags: -j8 -l8 SHELL=/nix/store/yjkch3aia9ny4dq42dbcjrdwqb1y8c33-bash-4.4-p23/bin/bash
Scanning dependencies of target looking-glass-client
[ 18%] Building C object CMakeFiles/looking-glass-client.dir/lg-fonts.c.o
[ 18%] Building C object CMakeFiles/looking-glass-client.dir/main.c.o
[ 18%] Building C object CMakeFiles/looking-glass-client.dir/ll.c.o
[ 18%] Building C object CMakeFiles/looking-glass-client.dir/lg-renderer.c.o
[ 22%] Building C object CMakeFiles/looking-glass-client.dir/utils.c.o
[ 27%] Building C object CMakeFiles/looking-glass-client.dir/spice/rsa.c.o
[ 31%] Building C object CMakeFiles/looking-glass-client.dir/spice/spice.c.o
[ 36%] Building C object CMakeFiles/looking-glass-client.dir/decoders/null.c.o
[ 40%] Building C object CMakeFiles/looking-glass-client.dir/decoders/yuv420.c.o
/build/source/client/spice/spice.c:39:10: fatal error: spice/error_codes.h: No such file or directory
 #include <spice/error_codes.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/looking-glass-client.dir/build.make:141: CMakeFiles/looking-glass-client.dir/spice/spice.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/looking-glass-client.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
builder for '/nix/store/ygr0f5giydw8virjy14vscllh4ncff6s-looking-glass-client-a12.drv' failed with exit code 2
cannot build derivation '/nix/store/ljm1wapslzwfs9zrai78fax9cf4s5v7s-system-path.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xjqrnrs0r8n4jwb0d2bsyja0mxr4zi44-nixos-system-nixos-workstation-19.09pre177651.aeb464dfd37.drv': 1 dependencies couldn't be built
error: build of '/nix/store/xjqrnrs0r8n4jwb0d2bsyja0mxr4zi44-nixos-system-nixos-workstation-19.09pre177651.aeb464dfd37.drv' failed
```


### Steps to reproduce
Add looking-glass-client package and rebuild. This used to work but something changed awhile back. I just commented out the package and forgot about it because I wasn't using it at the time. Now I'd like to use it again.


## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 5.0.10, NixOS, 19.09pre177651.aeb464dfd37 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre177651.aeb464dfd37"`
 - channels(nix): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

--- a/pkgs/applications/virtualization/looking-glass-client/default.nix
+++ b/pkgs/applications/virtualization/looking-glass-client/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub
+{ stdenv, fetchFromGitHub, fetchpatch
 , cmake, pkgconfig, SDL2, SDL, SDL2_ttf, openssl, spice-protocol, fontconfig
 , libX11, freefont_ttf, nettle, libconfig
 }:
@@ -19,6 +19,15 @@ stdenv.mkDerivation rec {
   buildInputs = [
     SDL SDL2 SDL2_ttf openssl spice-protocol fontconfig
     libX11 freefont_ttf nettle libconfig cmake
+  ];
+
+  patches = [
+    # Fix obsolete spice header usage. Remove with the next release. See https://github.com/gnif/LookingGlass/pull/126
+    (fetchpatch {
+      url = "https://github.com/gnif/LookingGlass/commit/2567447b24b28458ba0f09c766a643ad8d753255.patch";
+      sha256 = "04j2h75rpxd71szry15f31r6s0kgk96i8q9khdv9q3i2fvkf242n";
+      stripLen = 1;
+    })
   ];
 
   enableParallelBuilding = true;

